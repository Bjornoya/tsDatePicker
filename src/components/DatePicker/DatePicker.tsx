import React from 'react';
import ReactDatePicker from 'react-datepicker';
import { format } from 'date-fns';
import ru from 'date-fns/locale/ru';

import './DatePicker.scss';

export interface DatePickerProps {
    onChange: (event: any) => void
    date: Date
    dateFormat?: string
    label?: string
    disabled?: boolean
    invalid?: boolean
    placeholder?: string
    children?: React.ReactNode
}

const DatePicker = ({children, placeholder, disabled=false,
                        label, dateFormat="yyyy/MM/dd", date, onChange}: DatePickerProps) => {
    return (
        <div className="datepicker__wrapper">
            { label ? <label className="datepicker__label">{label}</label> : null }
            <ReactDatePicker
                locale={ru}
                selected={date}
                dateFormat={dateFormat}
                onChange={onChange}
                placeholderText={placeholder}
                disabled={disabled}
                todayButton={`Сегодня ${format(date, 'dd.MM.yyyy')}`}
            />
        </div>
    );
};

export default DatePicker;