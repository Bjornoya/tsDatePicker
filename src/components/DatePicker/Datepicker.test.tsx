import React from "react";
import { shallow, mount } from 'enzyme';
import DatePicker from "./DatePicker";

const onChange = jest.fn();
const date = new Date();

describe('DatePicker component', () => {
    let component: any;
    beforeEach(() => {
        component = shallow(<DatePicker onChange={onChange} date={date} label="Дата" placeholder="Введите дату" dateFormat="yyyy.MM.dd" />);
    });

    it('Renders DatePicker component', () => {
        expect(component).toMatchSnapshot();
    });

    describe('Default values', () => {

        it('Should have correct default disabled value', () => {
            const wrapper = component.find('t');
            expect(wrapper.prop('disabled')).toBe(false);
        });
        it('Renders correct default dateFormat value', () => {
            const component = shallow(<DatePicker onChange={onChange} date={date} />);
            const wrapper = component.find('t');
            expect(wrapper.prop('dateFormat')).toBe('yyyy/MM/dd');
        });
    });

    describe('Component Props', () => {
        it('Receives correct placeholder in props', () => {
            const wrapper = component.find('t');
            expect(wrapper.prop('placeholderText')).toBe('Введите дату');
        });
        it('Receives correct dateFormat in props', () => {
            const wrapper = component.find('t');
            expect(wrapper.prop('dateFormat')).toBe('yyyy.MM.dd');
        });
        it('Renders Label if have props', () => {
            const component = shallow(<DatePicker onChange={onChange} date={date} label="Дата"/>);
            const wrapper = component.find('label');
            expect(wrapper.text()).toEqual('Дата');
        });
        it('Receives correct disabled in props', () => {
            const component = shallow(<DatePicker onChange={onChange} date={date} disabled={true} />);
            const wrapper = component.find('t');
            expect(wrapper.prop('disabled')).toBe(true);
        });
    });

    describe('Component State', () => {
        it('State updates correctly', () => {
            const component = mount(<DatePicker onChange={onChange} date={date} placeholder="Введите дату" dateFormat="yyyy.MM.dd" />);
            component.find('input[type="text"]').simulate('change', {
                target: {
                    value: '2020/03/04',
                },
            });
            expect(component.find('input[type="text"]').prop('value')).toEqual(
                '2020/03/04',
            );
        });
    });

    describe('NOT Render', () => {
        const component = shallow(<DatePicker onChange={onChange} date={date} />);
        it('Not render label if props not received', () => {
            const wrapper = component.find('label');
            expect(wrapper.length).toEqual(0);
        })
    });

});