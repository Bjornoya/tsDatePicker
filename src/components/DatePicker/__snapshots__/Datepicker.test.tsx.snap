// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker component Renders DatePicker component 1`] = `ShallowWrapper {}`;
